monkey/infection_monkey/island_api_client/http_client.py

```diff
@@ -122,17 +125,15 @@ def _send_request(
         *args,
         **kwargs,
     ) -> requests.Response:
-        if self._api_url is None:
-            raise RuntimeError(
-                "HTTP client is not connected to the Island server,"
-                "establish a connection with 'connect()' before "
-                "attempting to send any requests"
-            )
-        url = f"{self._api_url}/{endpoint}".strip("/")
+        if self._server_url is None:
+            raise RuntimeError("HTTP client does not have a server URL set")
```

Comment:
This should probably be `ValueError`

Reply:
`ValueError` means that parameters were incorrect. In this case the state is incorrect, so either custom or `RuntimeError` it is